src/shared-components/QueryBrowser/QueryBrowser.tsx

```diff
@@ -0,0 +1,250 @@
+      const tickFormat = xAxisTickFormat(span);
+      return tickFormat(tick);
+    },
+    [xAxisTickFormat, span],
+  );
+
+  // Set a reasonable Y-axis range based on the min and max values in the data
+  const findMin = (series: GraphSeries): GraphDataPoint => _.minBy(series, 'y');
+  const findMax = (series: GraphSeries): GraphDataPoint => _.maxBy(series, 'y');
+  let minY: number = findMin(data.map(findMin))?.y ?? 0;
+  let maxY: number = findMax(data.map(findMax))?.y ?? 0;
+  if (minY === 0 && maxY === 0) {
+    minY = 0;
+    maxY = 1;
+  } else if (minY > 0 && maxY > 0) {
+    minY = 0;
+  } else if (minY < 0 && maxY < 0) {
+    maxY = 0;
+  }
+
+  domain.y = [minY, maxY];
+
+  return (
+    <div ref={containerRef} style={{ height: '500px' }}>
+      {(() => {
+        switch (true) {
+          case isLoading:
+            return <ChartSkeletonLoader />;
+          case !hasMetrics:
+            return <EmptyStateNoMetricsData />;
+          default: {
+            const labels: ChartVoronoiContainerProps['labels'] = ({
+              datum,
+            }) => {
+              const time = xTickFormat(datum.x);
+              let label = '';
+
+              if (metricsType === 'memory') {
+                label = `${datum?.style?.labels?.name}: ${yTickFormat(
+                  datum.y,
+                )} at ${xTickFormat(datum.x)}`;
+              } else if (metricsType === 'cpu') {
+                label = `${datum?.style?.labels?.name}: ${yTickFormat(
+                  datum.y,
+                )} at ${time}`;
+              }
+              return label;
+            };
+
+            return (
+              <Chart
+                ariaTitle={ariaTitle}
+                // containerComponent={graphContainer}
+                containerComponent={
+                  <ChartVoronoiContainer
+                    labels={labels}
+                    constrainToVisibleArea
+                  />
+                }
+                legendComponent={<ChartLegend data={legendData} />}
+                legendAllowWrap={true}
+                legendPosition="bottom-left"
+                scale={{ x: 'time', y: 'linear' }}
+                domain={domain}
+                height={chartHeight}
+                padding={chartPadding}
+                domainPadding={{ y: 20 }}
+                width={width}
+                themeColor={ChartThemeColor.multiUnordered}
+              >
+                <ChartAxis
+                  label={'\n' + t('axis_label_time')}
+                  tickCount={xAxisTickCount}
+                  tickFormat={xTickFormat}
+                />
+                <ChartAxis
+                  label={label}
+                  crossAxis={false}
+                  tickCount={6}
+                  dependentAxis
+                  tickFormat={yTickFormat}
+                />
+                <ChartGroup>
+                  {metricsType === 'memory'
+                    ? processedData.map((values, index) => {
```

Comment:
Add condition for array only , rest code keep same.

[metricsType === 'memory' ?  processedData: data].map((values, index)=>{
  //add code here
});

By doing this we can avoid to write duplicate code.